src/test/scala/scalapb/json4s/JsonFormatSpec.scala

```diff
@@ -343,6 +345,25 @@ class JsonFormatSpec extends FlatSpec with MustMatchers with OptionValues with J
     new Parser().fromJsonString[MyTest]("""{"opt_enum":2}""") must be(MyTest(optEnum = Some(MyEnum.V2)))
   }
 
+  "TestProto" should "infer default value when present" in {
+    new Parser().fromJsonString[MyTest]("""{"opt_enum":10}""") must be(MyTest(optEnum = Some(MyEnum.Unrecognized(10))))
```

Comment:
Can you move to the new EnumFormatSpec? Also add java-equivalence tests.

Reply:
Moved new Enum tests to EnumFormatSpec. Did not add java/scala failure symmetricity tests. Should I?

Reply:
Yes, the parsers should accept and reject the same inputs. Thank you!

Reply:
If you feel tests are not really clear let me know - I can try to re-factor them. Now I think coverage is good, just specifying all proto2/3 ignoringUnknownFields true/false variations is a bit tricky.